packages/webview_flutter/webview_flutter_wkwebview/lib/src/foundation/foundation.dart

```diff
@@ -267,7 +267,17 @@ class NSObject with Copyable {
 
   final NSObjectHostApiImpl _api;
 
-  /// Informs the observing object when the value at the specified key path has changed.
+  /// Informs the observing object when the value at the specified key path has
+  /// changed.
+  ///
+  /// {@template webview_flutter_wkwebview.foundation.callbacks}
+  /// For the associated Objective-C object to be automatically garbage
+  /// collected, it is required that this Function doesn't contain a strong
+  /// reference to the encapsulating class instance. Consider using
+  /// `WeakReference` when referencing an object not received as a parameter.
+  /// Otherwise, use [NSObject.dispose] to release the associated Objective-C
+  /// object manually.
```

Comment:
This explanation was for callbacks in general and not just KVO. I could write a separate one for KVO?

The use of `NSObject.dispose` was for the case that someone would want to guarantee a reference to an object until a specific callback was made. For example,

```dart
MyClass? myClass;

void aMethod() {
  myClass = MyClass(callbackWithImportantInfo: (String info) {
    if (myClass != null) {
      myClass.otherMethod();
      NSObject.dispose(myClass);
    }
    myClass = null;
  });
}

void aMethodCalledSometimeLater() {
  myOtherClass.setMyClass(myClass);
}
```

I think this could still apply to KVO. I agree that this use case would probably be rare though.